hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/Constants.java

```diff
@@ -1734,6 +1734,15 @@ private Constants() {
    */
   public static final boolean CHECKSUM_VALIDATION_DEFAULT = false;
 
+  /**
+   * Indicates the algorithm used to create the checksum for the object
+   * to be uploaded to S3. Unset by default. It supports the following values:
+   * 'CRC32', 'CRC32C', 'SHA1', and 'SHA256'
+   * value:{@value}
+   */
+  public static final String CHECKSUM_ALGORITHM =
```

Comment:
let's call this fs.s3a.create.checksum.algorithm, as there are lots of fs.s3a.create. items and it makes clear that this is what the checksum is exclusively for. We are having *a lot* of checksum problems with the 2.30 SDK and I don't want confusion